components/AddRecipe.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import {
+  View,
+  KeyboardAvoidingView,
+  Text,
+  StyleSheet,
+  TextInput,
+  TouchableHighlight,
+  Alert
+} from 'react-native';
+import { addRecipe } from '../services/RecipeService.js'
+
+export default class AddRecipe extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      text: '',
+      error: false
+    }
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+  
+  handleSubmit() {
```

Comment:
I believe if you do `handleSubmit = () => {` instead, it will bind it, and you don't have to do the bind in the constructor